Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,259 @@
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
+    }
+
+    /**
+     * Test the format parameter set in the class
+     */
+    public function testFormat()
+    {
+        $format = 'text/plain';
+        $this->driver->setFormat($format);
+        $this->assertEquals($format, $this->driver->getFormat());
+    }
+
+    /**
+     * @throws AMQPProtocolException
+     */
+    public function testConnectWithoutData()
+    {
+        $this->driver->declareChannel();
+        $consumer = $this->createConsumer();
+        $this->driver->destroy($consumer->getName());
+        $this->driver->configure('', '', '', '');
+        $this->driver->connect();
+        $this->assertTrue($this->driver->isConnected());
+        $this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
+    }
+
+    /**
+     * @dataProvider getMessages
+     */
+    public function testConsumeWaitNoBlock(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($message->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->waitNoBlock();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Prepare the data and class to consume a message
+     *
+     * @param MessageInterface $msg
+     * @return mixed
+     */
+    public function prepareToConsume(MessageInterface $msg)
+    {
+        $msgIn = $this->createQueueMessage($msg);
+        $msgIn->addHeader('test_header', '12345');
+        $this->driver->send($msgIn);
+        $this->driver->disconnect();
+        $this->driver->connect();
+        $this->consumer = $this->createConsumer();
+    }
+
+    /**
+     * Test that the QueueMessageInterface header compatible with AMQP Headers are sent to the queue and back to
+     * the consumer without changes.
+     */
+    public function testAMQPMessageHeaders()
+    {
+        $queueMessage = $this->createQueueMessage(new Entity());
+        $queueMessage->setTTL(rand(60, 600));
+        $queueMessage->setPriority(rand(0, 255));
+        $queueMessage->setMessageType(md5(rand(0, 255)));
+        $queueMessage->setPersistent(true);
+
+        $this->driver->send($queueMessage);
+        $this->consumer = $this->createConsumer();
```

Comment:
As in the rest of the of the code, mock the consumer shouldn't be needed

Reply:
Ok, it was removed.